client/src/nv_ingest_client/client/job_manager.py

```diff
@@ -0,0 +1,348 @@
+        """
+        if self._client is not None:
+            raise ValueError("self._client already exists.")
+
+        self._client = NvIngestClient()
+
+    def run(self, **kwargs: Any) -> List[Dict[str, Any]]:
+        """
+        Synchronously submits jobs to the NvIngestClient and fetches the results.
+
+        Parameters
+        ----------
+        kwargs : dict
+            Additional parameters for `submit_job` and `fetch_job_result` methods of NvIngestClient.
+
+        Returns
+        -------
+        List[Dict]
+            Result of each job after execution.
+        """
+        self._job_ids = self._client.add_job(self._job_specs)
+
+        submit_args = list(inspect.signature(self._client.submit_job).parameters)
+        submit_dict = {k: kwargs.pop(k) for k in dict(kwargs) if k in submit_args}
+        self._job_states = self._client.submit_job(self._job_ids, self._job_queue_id, **submit_dict)
+
+        fetch_args = list(inspect.signature(self._client.fetch_job_result).parameters)
+        fetch_dict = {k: kwargs.pop(k) for k in dict(kwargs) if k in fetch_args}
+        result = self._client.fetch_job_result(self._job_ids, **fetch_dict)
+
+        return result
+
+    def run_async(self, **kwargs: Any) -> Future:
+        """
+        Asynchronously submits jobs and returns a single future that completes when all jobs have finished.
+
+        Parameters
+        ----------
+        kwargs : dict
+            Additional parameters for the `submit_job_async` method.
+
+        Returns
+        -------
+        Future
+            A future that completes when all submitted jobs have reached a terminal state.
+        """
+        self._job_ids = self._client.add_job(self._job_specs)
+        future_to_job_id = self._client.submit_job_async(self._job_ids, self._job_queue_id, **kwargs)
+        self._job_states = {job_id: self._client._get_and_check_job_state(job_id) for job_id in self._job_ids}
+
+        combined_future = Future()
+        submitted_futures = set(future_to_job_id.keys())
+        completed_futures = set()
+        future_results = []
+
+        def _done_callback(future):
+            job_id = future_to_job_id[future]
+            job_state = self._job_states[job_id]
+            try:
+                result = self._client.fetch_job_result(job_id)
+                if job_state.state != JobStateEnum.COMPLETED:
+                    job_state.state = JobStateEnum.COMPLETED
+            except Exception:
+                result = None
+                if job_state.state != JobStateEnum.FAILED:
+                    job_state.state = JobStateEnum.FAILED
+            completed_futures.add(future)
+            future_results.append(result)
+            if completed_futures == submitted_futures:
+                combined_future.set_result(future_results)
+
+        for future in future_to_job_id:
+            future.add_done_callback(_done_callback)
+
+        return combined_future
+
```

Comment:
Thinking more about this, we're going to want an interface verb for something like `load()`, that checks to see if all the files specified exist locally, or need to be retrieved from some external source. It should be optional, because there will be some use cases where we want to let the ingest service itself do the load. 

We also are going to want a method like `files([....])`, that lets someone create the job_manager without selecting the file set. This will make the class re-usable, and a bit more flexible.

```python

job_manager.files(my_files)
result = job_managet.load().extract().run()